app/jobs/reminder_deadline_job.rb

```diff
@@ -1,13 +1,9 @@
 class ReminderDeadlineJob < ApplicationJob
   def perform
-    organizations = Organization.needs_reminding
+    remind_these_partners = Partners::FetchPartnersToRemindNowService.new.fetch
 
-    organizations.includes(:partners).each do |organization|
-      organization.partners.where(send_reminders: true).each do |partner|
-        unless partner.deactivated?
-          ReminderDeadlineMailer.notify_deadline(partner, organization).deliver_now
-        end
-      end
+    remind_these_partners.each do |partner|
+      ReminderDeadlineMailer.notify_deadline(partner).deliver_later
```

Comment:
I changed it to avoid having a long-running process handling each partner that needs to be `delivered`. This is because I am pretty sure `deliver_now` executes the delivery at the moment and `deliver_later` queues it to be run by workers. Furthermore, if one partner happens to trigger a raised error, it would make every other partner fail to receive the reminder.

Primarily this optimizes against having a single long-running job and a bit of fault tolerance in case something goes wrong.